src/node_http2.cc

```diff
@@ -893,32 +893,33 @@ int Http2Session::DoWrite(WriteWrap* req_wrap,
   return 0;
 }
 
-void Http2Session::AllocateSend(uv_buf_t* buf) {
-  buf->base = stream_alloc();
-  buf->len = kAllocBufferSize;
+size_t Http2Session::AllocateSend(WriteWrap** req) {
+  HandleScope scope(env()->isolate());
+  auto AfterWrite = [](WriteWrap* req, int status) {
+    req->Dispose();
+  };
+  Local<Object> obj =
+      env()->write_wrap_constructor_function()
+          ->NewInstance(env()->context()).ToLocalChecked();
+  // Base the amount allocated on the remote peers max frame size
+  uint32_t size =
+      nghttp2_session_get_remote_settings(
+          session(),
+          NGHTTP2_SETTINGS_MAX_FRAME_SIZE);
```

Comment:
This makes it 4x smaller than previously (which was equivalent to default window size + frame headers). Is that ok?

It could also potentially be allocating more than it needs to if the current available window size isn't very big.

Even if that's all fine, I think it needs `+ 9` (could define this elsewhere to be less magicky) because this doesn't account for the frame header.

Reply:
+1 on the +9. Going to keep benchmarking for a good allocation size